src/Repository/ContactRepository.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repository;
+
+use App\Dto\ContactRequest;
+use App\Entity\Contact;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class ContactRepository implements ContactRepositoryInterface
+{
+    private $entityManager;
+    private $repository;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+        $this->repository = $entityManager->getRepository(Contact::class);
+    }
+
+    public function save(Contact $contact): void
+    {
+        $this->entityManager->persist($contact);
+        $this->entityManager->flush();
+    }
+
+    public function remove(Contact $contact): void
+    {
+        $this->entityManager->remove($contact);
+        $this->entityManager->flush();
+    }
+
+    public function find(string $id): ?Contact
+    {
+        return $this->repository->find($id);
+    }
+
+    public function findAll(): array
+    {
+        return $this->repository->findAll();
+    }
+
+    public function createFromRequest(ContactRequest $contactRequest): Contact
+    {
+        $contact = new Contact();
+
+        return $contact
+            ->setId($this->nextIdentity()->toString())
+            ->setFirstName($this->firstName)
+            ->setLastName($this->lastName)
+            ->setEmail($this->email)
+            ->setPhoneNumber($this->phoneNumber)
+            ->setAddresses($this->addresses)
+        ;
+    }
+
+    public function createWith(
+        string $firstName,
+        string $lastName,
+        string $email,
+        string $phoneNumber,
+        string $addresses
+    ): Contact {
+        $contact = new Contact();
+
+        return $contact
```

Comment:
Same here, the best way, in my opinion, is to put all needed information into the constructor to have dependencies injection and allow your object to be immutable. If you pass all into the constructor and removed the unnecessary set() methods, then there is no chance to modify the object once created

Reply:
discussed below